### ISSUE TITLE
Text Editor: Crash on double clicking on an empty line

### ISSUE THREAD

**Issue by Kubiisek** [SerenityOS/serenity#5995]
Crash only happens in `.js`, `.cpp`, `.h` (these are the one I tested) files. Other files like `.frm`, `Makefile` and `.md` are fine.
Doesn't crash in HackStudio.
Stack trace:
```
CrashDaemon(18): New coredump file: /tmp/coredump/TextEditor_54_1616921537
CrashDaemon(18): --- Backtrace for thread #0 (TID 54) ---
CrashDaemon(18): 0x5acab1a8: [libsystem.so]  +0x8 (syscall.cpp:35)
CrashDaemon(18): 0xb1f6dcc6: [libc.so] __assertion_failed +0x106 (syscall.h:44)
CrashDaemon(18): 0x624f1b7f: [libgui.so] GUI::TextDocument::text_in_range(GUI::TextRange const&) const +0x2bf (Utf32View.h:91)
CrashDaemon(18): 0x624f73fe: [libgui.so] GUI::TextEditor::selected_text() const +0x7e (TextEditor.cpp:1235)
CrashDaemon(18): 0x812072ac: [/bin/TextEditor] TextEditorWidget::update_statusbar() +0x1cc (Atomic.h:266)
CrashDaemon(18): 0x8120768d: [/bin/TextEditor] AK::Function<void ()>::CallableWrapper<TextEditorWidget::TextEditorWidget()::{lambda()#17}>::call() const +0x1d (Function.h:111)
CrashDaemon(18): 0x624fb4b3: [libgui.so] GUI::TextEditor::set_cursor(GUI::TextPosition const&) +0xe3 (Function.h:57)
CrashDaemon(18): 0x624fbeed: [libgui.so] GUI::TextEditor::doubleclick_event(GUI::MouseEvent&) +0x1bd (TextEditor.cpp:256)
CrashDaemon(18): 0x6251ae9a: [libgui.so] .L2872 +0xa (Widget.cpp:271)
CrashDaemon(18): 0x78277c84: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(18): 0x62528368: [libgui.so] GUI::Window::handle_mouse_event(GUI::MouseEvent&) +0x5b8 (Window.cpp:375)
CrashDaemon(18): 0x6252aa19: [libgui.so] GUI::Window::event(Core::Event&) +0x39 (Window.cpp:545)
CrashDaemon(18): 0x78277c84: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(18): 0x78260ff4: [libcore.so] Core::EventLoop::pump(Core::EventLoop::WaitMode) +0x3e4 (Atomic.h:266)
CrashDaemon(18): 0x782615bb: [libcore.so] Core::EventLoop::exec() +0x4b (EventLoop.cpp:390)
CrashDaemon(18): 0x62438d65: [libgui.so] GUI::Application::exec() +0x25 (Application.cpp:121)
CrashDaemon(18): 0x81201696: [/bin/TextEditor] main +0x496 (main.cpp:106)
CrashDaemon(18): 0x81201a05: [/bin/TextEditor] _start +0x45 (crt0_shared.cpp:60)
```

**Comment by Kubiisek** [comment]
1. find: it looks like `m_editor::on_selection_change` inside `TextEditorWidget` is getting called for each selected character - not sure if that's the correct behavior.

**Comment by Kubiisek** [comment]
Ok, so the problem was in `TextDocument::text_in_range`. `selection_start_column_on_line` and `selection_end_column_on_line` will be the same and `Utf32View` doesn't like that. The obvious fix is to put there a check to see if those two are different and if so just return an empty String.

**Comment by NoHandsMate** [comment]
Maybe mark this al solved?

**Comment by linusg** [comment]
@NoHandsMate once #5997 is merged.

**Comment by NoHandsMate** [comment]
idk why my brain just thought that was merged. Sorry!

**Comment by linusg** [comment]
Fixed by #6537.
